main.go

```diff
@@ -111,8 +112,16 @@ func init() { //nolint:gochecknoinits
 	utilruntime.Must(securityv1.Install(scheme))
 }
 
-func initComponents(_ context.Context, p cluster.Platform) error {
-	return dashboardctrl.Init(p)
+func initComponents(_ context.Context, p cluster.Platform) *multierror.Error {
+	var multiErr *multierror.Error
+
+	if err := dashboardctrl.Init(p); err != nil {
+		multiErr = multierror.Append(multiErr, err)
+	}
+	if err := rayctrl.Init(p); err != nil {
+		multiErr = multierror.Append(multiErr, err)
+	}
+	return multiErr
```

Comment:
maybe use `ErrorOrNil()` here so it is consistent with the standard error pattern